pkg/reconciler/nscert/nscert.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nscert
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"regexp"
+	"text/template"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubelabels "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/sets"
+	kubelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/logging"
+	"knative.dev/serving/pkg/apis/networking"
+	"knative.dev/serving/pkg/apis/networking/v1alpha1"
+	listers "knative.dev/serving/pkg/client/listers/networking/v1alpha1"
+	"knative.dev/serving/pkg/network"
+	rbase "knative.dev/serving/pkg/reconciler"
+	"knative.dev/serving/pkg/reconciler/nscert/config"
+	"knative.dev/serving/pkg/reconciler/nscert/resources"
+)
+
+// Reconciler implements controller.Reconciler for Certificate resources.
+type reconciler struct {
+	*rbase.Base
+
+	// listers index properties about resources
+	nsLister            kubelisters.NamespaceLister
+	knCertificateLister listers.CertificateLister
+
+	configStore configStore
+}
+
+// Check that our Reconciler implements controller.Reconciler
+var _ controller.Reconciler = (*reconciler)(nil)
+var domainTemplateRegex *regexp.Regexp = regexp.MustCompile(`^\*\..+$`)
+
+// Reconciler implements controller.Reconciler for Namespace resources.
+func (c *reconciler) Reconcile(ctx context.Context, key string) error {
+	logger := logging.FromContext(ctx)
+	ctx = c.configStore.ToContext(ctx)
+
+	_, ns, err := cache.SplitMetaNamespaceKey(key)
+	namespace, err := c.nsLister.Get(ns)
+
+	if apierrs.IsNotFound(err) {
+		logger.Errorf("Namespace %s in work queue no longer exists %s", key, err)
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	if _, ok := namespace.Labels[networking.WildcardCertLabelKey]; ok {
```

Comment:
it also seems you're checking for `true/false` in the event handler; perhaps we need to check for actual value=true here?
basically `if m[k] =="true"` ?

Reply:
What I really need in the controller is to only include objects WITHOUT a given label. `LabelFilterFunc` with a value of `false` and `allowUnset = true` seemed like the easiest way. Maybe it would be better for me to just write `LabelDoesNotExistFilterFunc` or something like that.